apps/react-world/src/routes/AppRoutes.tsx

```diff
@@ -28,3 +37,17 @@ export const AppRoutes = () => {
     </Layout>
   );
 };
+
+// Path 변경에 따른 NavBar 상태 변경을 처리하기 위한 처리
+const NavBarWithLocation = () => {
+  const location = useLocation();
+
+  let selectedNavItem: NavItemType = 'home'; // default
+  for (const navItem of NAV_ITEMS) {
+    if (location.pathname === navItem.path) {
+      selectedNavItem = navItem.name;
+      break;
+    }
+  }
```

Comment:
let 변수와 for문을 통해 초기화하는 방식을 선택한 이유가 있으실까요?!

모든 언어에서도 동일하겠지만, 자바스크립트는 그 역사 때문에 특히나 스코프 바깥 변수를 변경하는 것에 민감합니다.
위 구문은 find로 대체가 가능할 것 같아요.

```suggestion
  const { name: selectedNavItem } =
    NAV_ITEMS.find(({path}) => location.pathname === path) || { name: 'home' };
```

Reply:
find 활용해서 이런 식으로 구조분해 할당이 가능하군요! 아래와 같이 수정했습니다 ㅎㅎ

```ts
const location = useLocation();

const { name: selectedNavItem } = NAV_ITEMS.find(
  navItem => location.pathname === navItem.path,
) || { name: 'home' };
return <Navbar selectedNavItem={selectedNavItem} />;
```